### ISSUE TITLE
open_at_time bug with offset -1 day

### ISSUE THREAD

**Issue by stopdesign** [rsheftel/pandas_market_calendars#164]
This is a schedule for CME_Equity. There should be one hour interval between market_close and market_open.
Dataframe looks good, but open_at_time returns more False than it should. It is more like 2 h break in this case:
```
                         market_open               break_start                 break_end              market_close
2022-01-10 2022-01-09 23:00:00+00:00 2022-01-10 21:15:00+00:00 2022-01-10 21:30:00+00:00 2022-01-10 22:00:00+00:00
2022-01-11 2022-01-10 23:00:00+00:00 2022-01-11 21:15:00+00:00 2022-01-11 21:30:00+00:00 2022-01-11 22:00:00+00:00

2022-01-10 21:00:00+00:00 True
2022-01-10 21:10:00+00:00 True
2022-01-10 21:20:00+00:00 False
2022-01-10 21:30:00+00:00 True
2022-01-10 21:40:00+00:00 True
2022-01-10 21:50:00+00:00 True
2022-01-10 22:00:00+00:00 False
2022-01-10 22:10:00+00:00 False
2022-01-10 22:20:00+00:00 False
2022-01-10 22:30:00+00:00 False
2022-01-10 22:40:00+00:00 False
2022-01-10 22:50:00+00:00 False
2022-01-10 23:00:00+00:00 False
2022-01-10 23:10:00+00:00 False
2022-01-10 23:20:00+00:00 False
2022-01-10 23:30:00+00:00 False
2022-01-10 23:40:00+00:00 False
2022-01-10 23:50:00+00:00 False
2022-01-11 00:00:00+00:00 True
2022-01-11 00:10:00+00:00 True
```
In this case I expect time from 23:00 to 23:59 to be `open`.

I've seen this bug only in some markets. It looks like markets with `offset 0` works fine.

Here is a test code:
```
from datetime import datetime, timedelta, timezone
import pandas_market_calendars


def main():

    cal = pandas_market_calendars.get_calendar("CME_Equity")
    schedule = cal.schedule(datetime(2022, 1, 10), datetime(2022, 1, 11))

    print(schedule, "\n")

    start = datetime(2022, 1, 10, 21).replace(tzinfo=timezone.utc)
    for i in range(20):
        cur_time = start + timedelta(minutes=i*10)
        print(cur_time, cal.open_at_time(schedule, cur_time, only_rth=False))


if __name__ == "__main__":
    main()
```

```
$ pip freeze
exchange-calendars==3.5
korean-lunar-calendar==0.2.1
numpy==1.22.0
pandas==1.3.5
pandas-market-calendars==3.2  <<<<<<
pyluach==1.3.0
python-dateutil==2.8.2
pytz==2021.3
six==1.16.0
toolz==0.11.2
```

**Comment by Stryder-Git** [comment]
Thanks for pointing this out and I actually already know why it's doing that.
My apologies because I should have found this when updating this package a few months ago, but I will fix it asap.

**Comment by stopdesign** [comment]
Thank you!

I have a new observation. There are incorrect results for NYSE postmarket:
```
cal = pandas_market_calendars.get_calendar("NYSE")
schedule = cal.schedule(datetime(2022, 1, 10), datetime(2022, 1, 11), start="pre", end="post")
```
Intervals between 00 and 01 should return True as it is valid postmarket time.
```
                                 pre               market_open              market_close                      post
2022-01-10 2022-01-10 09:00:00+00:00 2022-01-10 14:30:00+00:00 2022-01-10 21:00:00+00:00 2022-01-11 01:00:00+00:00
2022-01-11 2022-01-11 09:00:00+00:00 2022-01-11 14:30:00+00:00 2022-01-11 21:00:00+00:00 2022-01-12 01:00:00+00:00

2022-01-10 08:00:00+00:00 False
2022-01-10 08:10:00+00:00 False
2022-01-10 08:20:00+00:00 False
2022-01-10 08:30:00+00:00 False
2022-01-10 08:40:00+00:00 False
2022-01-10 08:50:00+00:00 False
2022-01-10 09:00:00+00:00 True
2022-01-10 09:10:00+00:00 True
...
2022-01-10 23:40:00+00:00 True
2022-01-10 23:50:00+00:00 True
2022-01-11 00:00:00+00:00 False
2022-01-11 00:10:00+00:00 False
...
```

**Comment by Stryder-Git** [comment]
Yes and that's for the same reason actually.

The method checks the index for the presence of the date, if it is in there, it only checks that row, if it is not,  it just returns False.
That causes this bug with schedules where market times start before or go across midnight. This was not in the tests and, unfortunately, I didn't realize this. 

Again, my apologies, but I will fix it (and add it to the tests) very soon!

If you have any other suggestions/criticism, I look forward to hearing them :)

**Comment by rsheftel** [comment]
This is fixed in #166
